Graph generators and SQLite Exposed coverage with tests and some fixes to Graph saving and generating

MuhammetSultanov opened the pull request:
1) Decided to remove support for Char in graph saving and generating for reasons below:
 - When saving string "A" in SQLite, it confuses it's type with char 'A'.
 - It is more efficient to save char as a string because it will take approximately same amount storage, but it will be easier to deserialize saved data from SQLite DB.
2) Covered with tests graph generators (Random with fixed weight 1, Random with random weight, Flower snark, Star directed, Star undirected).
3) Covered with tests Save, Get and Delete functions of graph from SQLite DB.
4) Covered with tests Save and Update state of theme functions in SQLite DB.
5) Added tests to check that generators create and connect nodes only from the graph itself, not from outer sources.

MuhammetSultanov committed:
add: Jacoco and testing init file

MuhammetSultanov committed:
add: graph generators test

Added 80 tests for graph generators (tests are dynamic to make less copies of similar tests)
Graph generators are fully covered.

MuhammetSultanov committed:
merge: develop to test

Need to merge develop branch with test branch to be up to date with the tested code

MuhammetSultanov committed:
remove: support for char type in graph

Decided to remove char type because it is incontinent, instead of it we will save chars as string

MuhammetSultanov committed:
add: SQLite Exposed test for graph and App config

Covered with tests graph Saving, Reading and Deleting.
Covered with test Saving and Updating App Theme state.

AlexandrKudrya commented on src/test/kotlin/dataTest/GraphGeneratorsTest.kt:
```diff
@@ -0,0 +1,137 @@
+            }
+        }
+    }
+}
```
Добавь тесты инвариантов

MuhammetSultanov replied:
Добавил

AlexandrKudrya commented on src/test/kotlin/dataTest/SQLiteExposedGraphTest.kt:
```diff
@@ -0,0 +1,162 @@
+                addVertice(7, 10)
+                addVertice(8, 9)
+                addVertice(9, 10)
+
```
Много для unit, не очень понятно зачем

MuhammetSultanov replied:
Уменьшил

AlexandrKudrya commented on src/test/kotlin/dataTest/SQLiteExposedGraphTest.kt:
```diff
@@ -0,0 +1,162 @@
+                addVertice("G", "J")
+                addVertice("H", "I")
+                addVertice("I", "J")
+            }
```
Многа

AlexandrKudrya commented on src/test/kotlin/dataTest/SQLiteExposedGraphTest.kt:
```diff
@@ -0,0 +1,162 @@
+                addVertice(7f, 10f)
+                addVertice(8f, 9f)
+                addVertice(9f, 10f)
+            }
```
Многа

AlexandrKudrya commented on src/test/kotlin/dataTest/SQLiteExposedGraphTest.kt:
```diff
@@ -0,0 +1,162 @@
+        fun `test serializeGraph function with Int`() {
+            val serializedGraph = serializeGraph(intGraph)
+            val expectedSerializedGraph =
+                "1:(2, 1.0);(4, 1.0)|2:(3, 1.0);(5, 1.0);(7, 1.0)|3:(4, 1.0);(5, 1.0)|4:|5:(8, 1.0);(6, 1.0)|6:(9, 1.0)|7:(8, 1.0);(10, 1.0)|8:(9, 1.0)|9:(10, 1.0)|10::10"
```
Заменить бы на генератор, а не чтобы просто строка в тестах лежала

AlexandrKudrya replied:
Ну типо если менять, то плохо

MuhammetSultanov replied:
Уменьшил текст чтобы было легче читать.

MuhammetSultanov committed:
add: invariant tests in generators

Added tests to check that generators create and connect nodes only from the graph itself, not from outer sources.
